### ISSUE TITLE
Regression on >2.4.11 releases

### ISSUE THREAD

**Issue by noam-almog** [etorreborre/specs2#338]
Hi Erik,

This is a simplified version of a matcher I wrote, it works up to 2.4.11, on all versions higher than it 12+ it will not work anymore.

```scala
List("key1" -> "val1") must not(contain(atLeast( be_===("key2" -> "val2") or be_===("key3" -> "val3")))) 
// <-- fails with List((key1,val1)) does not contain at least 1 correct value(..)
```

Removing the not operator will actually make this matcher match even though it's not correct.
```
List("key1" -> "val1") must contain(atLeast(checks = be_===("key2" -> "val2") or be_===("key3" -> "val3")))  
/// <-- matches
```

**Comment by etorreborre** [comment]
Thanks for reporting this (which can be further simplified to "`Seq(1) must contain(atLeast(2))` is passing"). This is fixed in `2.4.18-SNAPSHOT` or `3.0-20150307115016-00503e1`.

**Comment by noam-almog** [comment]
Thanks for the quick response.
I ran this on specs2 3.0 (not the snapshot) and the defect was not reproduced.
